### ISSUE TITLE
Decorator to call an action before deleting a service

### ISSUE THREAD

**Issue by FastGeert** [zero-os/0-robot#157]
A very common thing in templates will be the fact that after uninstalling a service, the service object itself will need to be deleted. 

Currently self.delete() creates issues as the action that is executing this will kill itself, making the action unable to succeed.

> Implementation hint: A nice way to delete after uninstalling could happen through the use of a decorator.
```python
class MyTemplate(...):
...
    @selfdestruct()
    def uninstall():
        ....
```

**Comment by zaibon** [comment]
Instead of decorating action that will delete the service, we should be able to bind an action to execute when delete is called. It make more sense for a user to just delete a service when he doesn't need it anymore.
And it's up to the template developer to mark which action needs to be executed to clean up the service before deletion

**Comment by zaibon** [comment]
fixed in #228
